### ISSUE TITLE
Add Xcode 14.0 beta 1

### ISSUE THREAD

**Issue by bielikb** [actions/runner-images#5702]
### Tool name

Xcode 14.0 beta 1

### Tool license

Apple, Inc.

### Add or update?

- [X] Add
- [ ] Update

### Desired version

14.0 beta 1

### Approximate size

7.3GB

### Brief description of tool

Xcode IDE

### URL for tool's homepage

developer.apple.com

### Platforms where you need the tool

- [ ] Azure DevOps
- [ ] GitHub Actions

### Virtual environments where you need the tool

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [X] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

```bash
no
```

### Are you willing to submit a PR?

no

**Comment by miketimofeev** [comment]
Hi @bielikb!
We will deploy the new image with Xcode 14 beta later this week.

**Comment by bielikb** [comment]
Hi @miketimofeev,


thanks a lot for a prompt response and I'll keep watching this issue :)

**Comment by li-bei** [comment]
Hi @miketimofeev,

Xcode 14 doesn't include watchOS SDK and tvOS SDK by default, will new image includes all SDKs?

**Comment by miketimofeev** [comment]
@li-bei thanks for the heads-up, we haven't noticed it. Most likely not, but we will investigate

**Comment by li-bei** [comment]
@miketimofeev Thank you for the quick response.

I'd suggest pre-download watchOS SDK and tvOS SDK for new images, otherwise users may need download ~3GB size SDK every time.

**Comment by aaroncrespo** [comment]
@miketimofeev agree with Bei Li, if the environment doesn't come pre-populated with SDK's, as they historically have, users will need to devise a hosting solution themselves to avoid downloading several GB runtimes for their macOS jobs. 

For example the download to support iOS 14.0.1 is nearly 6GB compressed. And `xcodebuild` currently has no mechanism to only download specific SDK's via CLI.


<img width="799" alt="Screen Shot 2022-06-09 at 9 46 32 AM" src="https://user-images.githubusercontent.com/431662/172862467-47b75a28-86e3-440c-8cdd-d90bf01607f6.png">

**Comment by zorgiepoo** [comment]
> @li-bei @aaroncrespo thanks for the information. Unfortunately, we have faced another issue that was brought by Xcode 14 — simulators can't be installed via `xcrun simctl create 'Test simulator iPhone 8 (13.0)' 'iPhone 8' 'iOS13.0'` command using Xcode 13.* version with the error:
> `Unable to create a device for device type: iPhone 8 (com.apple.CoreSimulator.SimDeviceType.iPhone-8), runtime: iOS 13.0 (13.0 - 17A577) - com.apple.CoreSimulator.SimRuntime.iOS-13-0 (unavailable, runtime path not found)`
> We are currently investigating the issue and decided to postpone Xcode 14 beta rollout until we got some results.

I don't understand this how this particular example is a new problem. Xcode 14 comes with `iOS16.0` not `iOS13.0` which was trying to be installed here. Similar to how Xcode 13 comes with versions of iOS 15.x, which are the only ones listed on the [macos-12-Readme](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md). This example isn't using tvOS or watchOS too. This same command fails for me on Xcode 13.4 too.

**Comment by miketimofeev** [comment]
@Brett-Best I'm afraid we are still investigating and can not provide any ETA

**Comment by miketimofeev** [comment]
@bielikb sure! We will give it a try. No need to create a new ticket.

**Comment by miketimofeev** [comment]
@bielikb I'm afraid the same issue in our tests with beta 2. We will continue our investigation
```
An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=401):
The iOS 13.0 simulator runtime is not available.
runtime path not found
Download the iOS 13.0 simulator runtime from the Components section in Xcode's Preferences.
Unable to create a device for device type: iPhone 8 (com.apple.CoreSimulator.SimDeviceType.iPhone-8), runtime: iOS 13.0 (13.0 - 17A577) - com.apple.CoreSimulator.SimRuntime.iOS-13-0 (unavailable, runtime path not found)
```

**Comment by tytan652** [comment]
Can you explain why did you try to install iOS 13 simulator ?
macOS 12 images seems to have only simulators for iOS 15 and later.

https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md#installed-simulators

**Comment by vsafonkin** [comment]
> Can you explain why did you try to install/use a iOS 13 simulator ? macOS 12 image seems to have only simulators for iOS 15 and later.
> 
> https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md#installed-simulators

Hi @tytan652, we need to be sure that our customers would be able to install additional iOS SDKs and use it.

**Comment by Brett-Best** [comment]
Would it be possible to create an macOS 13 beta image with the Xcode 14 betas?

**Comment by ggoraa** [comment]
Are there any news yet?

**Comment by miketimofeev** [comment]
> Would it be possible to create an macOS 13 beta image with the Xcode 14 betas?

It would be, but no ETA, I'm afraid.

**Comment by ggoraa** [comment]
Guys, there is already Xcode 14 Beta 3, are there any ETAs about Xcode 14 arriving at lease on `macos-12` images?

**Comment by erikng** [comment]
Why can't you just install xcode 14 with the simulators it comes bundled with for now? Something is more ideal than **nothing**.

I'd really like to build a new version of my application through github and I am blocked by this.

**Comment by Brett-Best** [comment]
@miketimofeev would it be possible to create a new macOS 12 image (with a different agent name) with this as a known issue? Then only deploy a limited amount for the people that want to opt into it.

It would be great to get macOS 13 ASAP, though not expecting much given how long macOS 11 and 12 took.

**Comment by miketimofeev** [comment]
@Brett-Best I'm afraid it's quite a complicated process. We are currently trying to find a workaround.

**Comment by erikng** [comment]
> @erikng the problem is that the simulators for any other Xcode versions can't be installed after Xcode 14 is added to the image. In our pre-production set of tests, we use Xcode 13.2.1 and install ios 13 simulator (but the issue happens with the newer ios versions as well) using the `sudo xcversion simulators --install='iOS 13.0'` and `xcrun simctl create 'Test simulator iPhone 8 (13.0)' 'iPhone 8' 'iOS13.0'` commands and get an error on the second command:
> 
> ```
> Successfully installed iOS 13.0 Simulator
> Installing iPhone 8 test simulator
> Invalid runtime: iOS13.0
> ##[error]Bash exited with code '145'.
> ```

do you think it has to do with

```
Fixed: After installing Xcode 14 beta 3, Xcode 13.3.1 and 13.4.1 no longer show their respective iOS Simulators. Relaunch Xcode 14 beta 3 and bring up the Platforms Preference pane. Afterwards, you can relaunch the older Xcode to get the correct Simulators. (96100752)
```

from: https://developer.apple.com/documentation/Xcode-Release-Notes/xcode-14-release-notes

**Comment by PatTheMav** [comment]
Support for Xcode 14 would be highly appreciated given new Frameworks (and also new functionality added to existing Frameworks) which are only available with new SDK versions available on the runners.

Right now I'm implementing features available on macOS 12.4+ and macOS 13 respectively, but cannot provide (automatic) nightly builds for testing as neither is available on runners.

**Comment by leotm** [comment]
i know you guys working hard investigating/debugging

just noting CircleCI not supporting macOS 13 yet either
but Xcode 14 since [last month (b2)](https://discuss.circleci.com/t/xcode-14-beta-3-released/44690) and [couple wks ago (b3)](https://discuss.circleci.com/t/xcode-14-beta-3-released/44690) [(VM manifest)](https://circle-macos-docs.s3.amazonaws.com/image-manifest/v8405/index.html)
with known issues noted

similar with BitRise but with macOS w 13.0 SDKs, on [Intel](https://github.com/bitrise-io/bitrise.io/blob/master/system_reports/MACOS/INTEL/osx-xcode-14.0.x.log) and [M1](https://github.com/bitrise-io/bitrise.io/blob/master/system_reports/MACOS/M1/osx-xcode-14.0.x.log)

keep up the good work

**Comment by Kiruel** [comment]
Have you got any news on this one ? (btw Xcode 14 beta 3 is available)

**Comment by sjavora** [comment]
It's beta 4 now in fact...

**Comment by miketimofeev** [comment]
The issue still persists in beta 4 as well

**Comment by enricenrich** [comment]
@miketimofeev Temporarily, can't you do like CircleCI ([ref.](https://github.com/actions/virtual-environments/issues/5702#issuecomment-1195445174)) and offer the Xcode 14 Beta image with its current limitation? That would already unblock many of us for now. It's better this than nothing... 😕

**Comment by PatTheMav** [comment]
Not to put too fine a point on it, but the oldest simulator made available in Xcode 14 targets iOS 13.7. Given Apple's observed behaviour of focusing support on the current version plus prior two versions (which in this case would only include iOS 16, 15, and 14) it wouldn't come as a shock to me that iOS 13 is not considered a "supported" OS anymore and only the most recent simulator is made available as a nice gesture (and `xcrun simctl create 'Test simulator iPhone 8 (13.7)' 'iPhone 8' 'iOS13.7'` works just fine).

If people still need to test/build against iOS 13 (Apple most probably would rather have them focus their efforts on iOS 16), they'd still have Xcode 13-based images available to them.

PS: Do note that iOS 13 simulators used to have lots of issues running on M1 machines, so I wouldn't be surprised if Apple is more than happy to drop support for anything that doesn't play well with M1.

**Comment by leotm** [comment]
nice ^ for React Native we deprecated 11.0 for 12.4 in [March](https://github.com/facebook/react-native/commit/982ca30de079d7e80bd0b50365d58b9048fb628f), getting 13.7 on b3 so time to upg to b4

**Comment by zorgiepoo** [comment]
> This is from Xcode 14b4, you can add all the way back to iOS 12.4.

For me, it doesn't show anything below iOS 13.7
<img width="942" alt="Screenshot 2022-08-01 at 6 23 06 PM" src="https://user-images.githubusercontent.com/857267/182271710-a876458c-347c-423b-9408-c9907c358f77.png">
.

**Comment by zorgiepoo** [comment]
My other Mac (which is on Monterey (instead of Ventura) and also has Xcode 13.4 installed) shows me more Simulator runtimes down to iOS 12.4. If I download iOS 13.0 Simulator using Xcode 14 (beta 4)'s UI, I can later run:

```
xcrun simctl create 'Test simulator iPhone 8 (13.0)' 'iPhone 8' 'iOS13.0'
```

Without any issue over here. @miketimofeev is using `xcversions` to install the Simulator instead which I think is the only difference in these setups. (I don't have `xcversions` installed and it doesn't appear to be a standard tool).

**Comment by ggoraa** [comment]
![image](https://user-images.githubusercontent.com/61655029/182367400-e3ea5de8-53bd-409d-97ee-c58309dc3aa0.png)
I think this info useful to the issue

Got from release notes of Xcode 14 beta 4

**Comment by PatTheMav** [comment]
> > This is from Xcode 14b4, you can add all the way back to iOS 12.4.
> 
> For me, it doesn't show anything below iOS 13.7 <img alt="Screenshot 2022-08-01 at 6 23 06 PM" width="942" src="https://user-images.githubusercontent.com/857267/182271710-a876458c-347c-423b-9408-c9907c358f77.png"> .
> 
> Maybe it shows more options if you've older versions of Xcode installed.

I'm happy to be wrong and you could be right that this might depend on older versions being installed in tandem - with just Xcode 14b4 installed 13.7 is the oldest iOS simulator available on my machine, but that one's also on a macOS 13 already.

**Comment by A30008228-christian** [comment]
> @bielikb I'm afraid the same issue in our tests with beta 2. We will continue our investigation
> 
> ```
> An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=401):
> The iOS 13.0 simulator runtime is not available.
> runtime path not found
> Download the iOS 13.0 simulator runtime from the Components section in Xcode's Preferences.
> Unable to create a device for device type: iPhone 8 (com.apple.CoreSimulator.SimDeviceType.iPhone-8), runtime: iOS 13.0 (13.0 - 17A577) - com.apple.CoreSimulator.SimRuntime.iOS-13-0 (unavailable, runtime path not found)
> ```

This is called out as solved in the latest Xcode 14 beta 5 release notes  that dropped yesterday.
https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes
<img width="963" alt="image" src="https://user-images.githubusercontent.com/85143085/183562761-5d093958-a86b-4ce8-b2db-54e65117cf2b.png">
Would you be able to confirm?

**Comment by al-cheb** [comment]
The new macOS-12 image with Xcode Beta5 has been deployed - https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

![image](https://user-images.githubusercontent.com/47745270/183933492-a6cafbd4-5874-4647-84f7-85530059b203.png)

**Comment by Brett-Best** [comment]
@al-cheb would it be possible to pre-install the tvOS and watchOS runtimes? They are no longer included by default.

**Comment by al-cheb** [comment]
@Brett-Best, please create a new issue which versions should be included by default.

**Comment by Brett-Best** [comment]
@al-cheb https://github.com/actions/runner-images/issues/6051 hopefully it makes sense (:

**Comment by erikng** [comment]
Awesome news!
